integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/InternalEventHandlerFactory.java

```diff
@@ -155,7 +156,8 @@ public InternalEventHandlerFactory() {
         .add(
             new ErrorFacetBuilder(),
             new LogicalPlanRunFacetBuilder(context),
-            new SparkVersionFacetBuilder(context))
+            new SparkVersionFacetBuilder(context),
+            new EnvironmentFacetBuilder(context))
```

Comment:
Add the `System.getenv` check here - there's no point in checking that env variable for every listener event; either it's set or it isn't